source/taskmanager-client/src/main/java/nl/aerius/taskmanager/client/TaskManagerClientSender.java

```diff
@@ -68,142 +60,102 @@ public TaskManagerClientSender(final BrokerConnectionFactory factory) {
+        propertiesBuilder.deliveryMode(DELIVERY_MODE_PERSISTENT);
+
+        channel.get().basicPublish("", queueName, propertiesBuilder.build(), QueueHelper.objectToBytes(data));
         // task has been send successfully, return.
         done = true;
-      } catch (final ConnectException | TimeoutException e) {
+      } catch (final ConnectException e) {
         // don't catch all IOExceptions, just ConnectExceptions.
         // exceptions like wrong host-name should cause a bigger disturbance.
         // those indicate that connection has temporarily been lost with RabbitMQ, though could be not that temporarily...
-        LOG.error("Sending task (id: {}) failed, retrying in a bit.", wrapper.getTaskId(), e);
+        LOG.error("Sending task (id: {}) failed, retrying in a bit.", correlationId, e);
         try {
           Thread.sleep(CLIENT_START_RETRY_PERIOD);
         } catch (final InterruptedException e1) {
           // no need to log.
           Thread.currentThread().interrupt();
         }
-        LOG.info("Retrying to send task (id: {})", wrapper.getTaskId());
+        LOG.info("Retrying to send task (id: {})", correlationId);
       }
     }
   }
 
-  private Connection getConnection() {
-    return factory.getConnection();
-  }
-
   /**
-   * Starts a consumer, but only if a callback is specified.
-   * @param taskWrapper task data
-   * @return name of the reply queue
+   * Method to enrich the builder with additional data before sending the task.
+   *
+   * @param builder builder to enrich
    * @throws IOException
    */
-  private String startConsumer(final TaskWrapper taskWrapper) throws IOException {
-    final Channel channel = getConnection().createChannel();
-    final String replyQueueName = channel.queueDeclare().getQueue();
-    final TaskResultConsumer resultConsumer = new TaskResultConsumer(channel, taskWrapper, this);
-    channel.basicConsume(replyQueueName, true, resultConsumer);
-    return replyQueueName;
+  protected void prepareBeforeSend(final Builder builder) throws IOException {
+    // Default implementation does nothing.
+  }
+
+  protected boolean checkChannel(final AtomicReference<Channel> channelReference) throws IOException {
+    final Channel channel = channelReference.get();
+    if (channel == null || !channel.isOpen()) {
+      channelReference.set(getConnection().createChannel());
+      return true;
+    }
+    return false;
+  }
+
+  private Connection getConnection() {
+    return factory.getConnection();
   }
 
   /**
    * Exit this client, making it impossible to send new tasks or retrieve any more results.
    * Should be used by a TaskResultCallback when all the results are in (to free up resources).
    */
-  public void shutdown() {
+  @Override
+  public void close() {
     synchronized (this) {
       running = false;
       if (factory != null) {
         factory.deRegisterClient(this);
       }
+      closeChannel(channel.get());
```

Comment:
might want to check on `running` in `checkChannel` to prevent opening another channel again?